### PR TITLE
Add endpoint for ISO datastores

### PR DESCRIPTION
**Pull request by lfu**
Fixes https://github.com/ManageIQ/manageiq-api/issues/909

@miq-bot add_label enhancement, kasparov/no

### CODE REVIEW

File: spec/requests/iso_datastores_spec.rb [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+      expect(response).to have_http_status(:ok)
+    end
+  end
+
+  describe 'DELETE /api/iso_datastores/:id' do
+    it "deletes an iso datastore with an appropriate role" do
+      api_basic_authorize(action_identifier(:iso_datastores, :delete))
+      delete(api_iso_datastore_url(nil, iso_datastore))
+      expect(response).to have_http_status(:no_content)
```
Comment (abellotti) [inline spec/requests/iso_datastores_spec.rb:102]: maybe also check that it's physically deleted here too.

**Comment by lfu** [comment]
```
GET /api/iso_datastores

{
    "name": "iso_datastores",
    "count": 1,
    "subcount": 1,
    "pages": 1,
    "resources": [
        {
            "href": "http://localhost:3000/api/iso_datastores/10000000000001"
        }
    ],
    "actions": [
        {
            "name": "query",
            "method": "post",
            "href": "http://localhost:3000/api/iso_datastores"
        },
        {
            "name": "create",
            "method": "post",
            "href": "http://localhost:3000/api/iso_datastores"
        },
        {
            "name": "delete",
            "method": "post",
            "href": "http://localhost:3000/api/iso_datastores"
        }
    ],
    "links": {
        "self": "http://localhost:3000/api/iso_datastores?offset=0",
        "first": "http://localhost:3000/api/iso_datastores?offset=0",
        "last": "http://localhost:3000/api/iso_datastores?offset=0"
    }
}
```

```
GET /api/iso_datastores/10000000000001

{
    "href": "http://localhost:3000/api/iso_datastores/10000000000001",
    "id": "10000000000001",
    "ems_id": "10000000000024",
    "last_refresh_on": "2021-01-21T17:34:37Z",
    "actions": [
        {
            "name": "delete",
            "method": "post",
            "href": "http://localhost:3000/api/iso_datastores/10000000000001"
        },
        {
            "name": "delete",
            "method": "delete",
            "href": "http://localhost:3000/api/iso_datastores/10000000000001"
        }
    ]
}
```

```
POST /api/iso_datastores
{
    "action": "query",
    "resources": [
        { "href" : "http://localhost:3000/api/iso_datastores/10000000000001"}
    ]
}

{
    "results": [
        {
            "href": "http://localhost:3000/api/iso_datastores/10000000000001",
            "id": "10000000000001",
            "ems_id": "10000000000024",
            "last_refresh_on": "2021-01-21T17:34:37Z",
            "actions": [
                {
                    "name": "delete",
                    "method": "post",
                    "href": "http://localhost:3000/api/iso_datastores/10000000000001"
                },
                {
                    "name": "delete",
                    "method": "delete",
                    "href": "http://localhost:3000/api/iso_datastores/10000000000001"
                }
            ]
        }
    ]
}
```

```
POST /api/iso_datastores
{
    "ems_id": "10000000000024"
}

{
    "results": [
        {
            "href": "http://localhost:3000/api/iso_datastores/10000000000002",
            "id": "10000000000002",
            "ems_id": "10000000000024",
            "last_refresh_on": null
        }
    ]
}
```

```
POST /api/iso_datastores/10000000000002
{
    "action": "delete"
}

{
    "success": true,
    "message": "iso_datastores id: 10000000000002 deleting",
    "href": "http://localhost:3000/api/iso_datastores/10000000000002"
}
```

```
DELETE /api/iso_datastores/10000000000002
```

File: spec/requests/iso_datastores_spec.rb (hunk 1 above)
Comment (lfu) [reply spec/requests/iso_datastores_spec.rb:106]: Updated.

**Review by abellotti — APPROVED** [review]

**Comment by lfu** [comment]
Travis failed due to widget specs which is not related.

**Review by gtanzillo — APPROVED** [review]

**Pull request merged**